lib/srv/sess_test.go

```diff
@@ -988,3 +993,176 @@ func TestSessionRecordingMode(t *testing.T) {
 		})
 	}
 }
+
+func TestCloseProxySession(t *testing.T) {
+	srv := newMockServer(t)
+	srv.component = teleport.ComponentProxy
+
+	reg, _ := NewSessionRegistry(SessionRegistryConfig{
+		Srv:                   srv,
+		SessionTrackerService: srv.auth,
+	})
+	t.Cleanup(func() { reg.Close() })
+
+	scx := newTestServerContext(t, reg.Srv, nil)
+	sid := string(rsession.NewID())
+	scx.SetEnv(sshutils.SessionEnvVar, sid)
+
+	// Open a new session
+	sshChanOpen := newMockSSHChannel()
+	// Always close the session from the client side to avoid it being stuck
+	// on closing (server side).
+	t.Cleanup(func() { sshChanOpen.Close() })
+	go func() {
+		// Consume stdout sent to the channel
+		io.ReadAll(sshChanOpen)
+	}()
+
+	err := reg.OpenSession(context.Background(), sshChanOpen, scx)
+	require.NoError(t, err)
+	require.NotNil(t, scx.session)
+
+	// After the session is open, we force a close comming from the server. Do
+	// this inside a goroutine to avoid being blocked.
+	closeChan := make(chan error)
+	go func() {
+		closeChan <- scx.session.Close()
+	}()
+
+	select {
+	case err := <-closeChan:
+		require.NoError(t, err)
+	case <-time.After(10 * time.Second):
+		require.Fail(t, "expected session to be closed")
+	}
+}
+
+// TestClodeRemoteSession given a remote session recording at proxy ensure that
+// closing the session releases all the resources, and return properly to the
+// user.
+func TestCloseRemoteSession(t *testing.T) {
+	srv := newMockServer(t)
+	srv.component = teleport.ComponentProxy
+
+	// init a session registry
+	reg, _ := NewSessionRegistry(SessionRegistryConfig{
+		Srv:                   srv,
+		SessionTrackerService: srv.auth,
+	})
+	t.Cleanup(func() { reg.Close() })
+
+	scx := newTestServerContext(t, reg.Srv, nil)
+	scx.SessionRecordingConfig.SetMode(types.RecordAtProxy)
+	scx.RemoteSession = mockSSHSession(t)
+
+	sid := string(rsession.NewID())
+	scx.SetEnv(sshutils.SessionEnvVar, sid)
+
+	// Open a new session
+	sshChanOpen := newMockSSHChannel()
+	// Always close the session from the client side to avoid it being stuck
+	// on closing (server side).
+	t.Cleanup(func() { sshChanOpen.Close() })
+	go func() {
+		// Consume stdout sent to the channel
+		io.ReadAll(sshChanOpen)
+	}()
+
+	err := reg.OpenSession(context.Background(), sshChanOpen, scx)
+	require.NoError(t, err)
+	require.NotNil(t, scx.session)
+
+	// After the session is open, we force a close comming from the server. Do
+	// this inside a goroutine to avoid being blocked.
+	closeChan := make(chan error)
+	go func() {
+		closeChan <- scx.session.Close()
+	}()
+
+	select {
+	case err := <-closeChan:
+		require.NoError(t, err)
+	case <-time.After(10 * time.Second):
+		require.Fail(t, "expected session to be closed")
+	}
+}
+
+func mockSSHSession(t *testing.T) *tracessh.Session {
```

Comment:
`t.Helper()`

Reply:
you might have added `t.Helper()` in the wrong function =D

Reply:
True, I've updated it. Thanks!